packages/create-project/src/index.ts

```diff
@@ -1,151 +1,181 @@
 #!/usr/bin/env node
-import os from 'os';
+
+// Modified from: https://github.com/vercel/next.js/blob/2425f4703c4c6164cecfdb6aa8f80046213f0cc6/packages/create-next-app/index.ts
+
+import chalk from 'chalk';
+import Commander from 'commander';
 import path from 'path';
-import { mkdir, writeFile, readFile } from 'fs-extra';
-import arg from 'arg';
-import { spawn } from './subprocess';
-
-const command = '@temporalio/create';
-const typescriptVersion = '4.4.2';
-const nodeMajorVersion = parseInt(process.versions.node, 10);
-const npm = /^win/.test(process.platform) ? 'npm.cmd' : 'npm';
-
-const packageJsonBase = {
-  version: '0.1.0',
-  private: true,
-  scripts: {
-    build: 'tsc --build',
-    'build.watch': 'tsc --build --watch',
-    start: 'ts-node src/worker.ts',
-    'start.watch': 'nodemon src/worker.ts',
-    workflow: 'ts-node src/exec-workflow.ts',
-  },
-  devDependencies: {
-    typescript: `^${typescriptVersion}`,
-    [`@tsconfig/node${nodeMajorVersion}`]: '^1.0.0',
-    'ts-node': '^10.2.1',
-    nodemon: '^2.0.12',
-  },
-  nodemonConfig: {
-    watch: ['src'],
-    ext: 'ts',
-    execMap: {
-      ts: 'ts-node',
-    },
-  },
-};
-
-const tsConfig = {
-  extends: `@tsconfig/node${nodeMajorVersion}/tsconfig.json`,
-  version: typescriptVersion,
-  compilerOptions: {
-    emitDecoratorMetadata: false,
-    experimentalDecorators: false,
-    declaration: true,
-    declarationMap: true,
-    sourceMap: true,
-    composite: true,
-    rootDir: './src',
-    outDir: './lib',
-  },
-  include: ['src/**/*.ts'],
-  exclude: ['node_modules'],
-};
-
-/**
- * Copy sample from `source` to `target` stripping away snipsync comments
- */
-async function copySample(source: string, target: string) {
-  const code = await readFile(source, 'utf8');
-  const stripped = code.replace(/.*@@@SNIP(START|END).*\n/gm, '');
-  await writeFile(target, stripped);
-}
+import prompts from 'prompts';
+import checkForUpdate from 'update-check';
+import { createApp } from './create-project';
+import { shouldUseYarn } from './helpers/should-use-yarn';
+import { validateNpmName } from './helpers/validate-pkg';
+import packageJson from '../package.json';
 
-async function writePrettyJson(path: string, obj: any) {
-  await writeFile(path, JSON.stringify(obj, null, 2) + os.EOL);
-}
+const program = new Commander.Command(packageJson.name)
+  .version(packageJson.version, '-v, --version')
+  .arguments('<project-directory>')
+  .usage(`${chalk.green('<project-directory>')} [options]`)
+  .option(
+    '--use-yarn',
```

Comment:
This option is disregarded in most of the code here, consider dropping it.

Reply:
When the option is included, `yarn install` is used, a `yarn.lock` file is created, and the suggested commands use `yarn`. I don't know anyone with a strong yarn preference, but I imagine they're out there, in which case them not having to delete the `package-lock.json` file and run `yarn` themselves would be better UX? 

![image](https://user-images.githubusercontent.com/251288/132377710-ec90c7eb-c8fd-4852-be6c-9e6adefdf8b1.png)

Reply:
But I think `shouldUseYarn` doesn't check the flag and the `notifyUpdate` will show an npm command